resource datadog_service_definition_yaml throwing error

### Datadog Terraform Provider Version

v3.49.0

### Terraform Version

1.10.4

### What resources or data sources are affected?

datadog_service_definition_yaml

### Terraform Configuration Files

```hcl
resource "datadog_service_definition_yaml" "main" {
  count = var.environment == "staging" ? 1 : 0
  service_definition = <<EOF
schema-version: v2.2
dd-service: ${local.name}
team: ${local.team_name}
contacts:
  - name: #Email-SRE-DL
    type: email
    contact: redacted_email_address
  - name: #Email-Dev-DL
    type: email
    contact: redacted_email_address
type: web
lifecycle: production
languages:
  - ruby
links:
  - name: Confluence Docs
    type: doc
    provider: confluence
    link: redacted_confluence_uri
EOF
}
```

### Relevant debug or panic output

```
Stack trace from the terraform-provider-datadog_v3.49.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 139 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.getNameField(...)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_service_definition_yaml.go:235
github.com/terraform-providers/terraform-provider-datadog/datadog.normalizeArrayField.func1(0x4000492718?, 0x0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_service_definition_yaml.go:224 +0x1b0
sort.insertionSort_func({0x4000492838?, 0x400076f380?}, 0x0, 0x2)
	sort/zsortfunc.go:12 +0xc0
sort.stable_func({0x4000492838?, 0x400076f380?}, 0x2)
	sort/zsortfunc.go:343 +0x6c
sort.SliceStable({0x10f8b00?, 0x4000361398?}, 0x4000492838)
	sort/slice.go:44 +0xa8
github.com/terraform-providers/terraform-provider-datadog/datadog.normalizeArrayField(0x400076f2c0, {0x15437c2, 0x8})
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_service_definition_yaml.go:223 +0xec
github.com/terraform-providers/terraform-provider-datadog/datadog.prepServiceDefinitionResource(0x400076f2c0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_service_definition_yaml.go:134 +0x100
github.com/terraform-providers/terraform-provider-datadog/datadog.Provider.resourceDatadogServiceDefinitionYAML.func48.1({0x1144c60?, 0x400093a530?})
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_service_definition_yaml.go:92 +0x44
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffString(0x400000e5d1?, {0x1550565, 0x12}, 0x40006857c0, 0x4000492c48, {0x17f4120, 0x4000693280}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/schema.go:1549 +0xc0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0x400076f020, {0x17ee890, 0x400076eb10}, {0x1550565, 0x12}, 0x40006857c0, 0x4000693200, {0x17f4120, 0x4000693280}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/schema.go:1128 +0xdc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0x400076f020, {0x17ee890, 0x400076eb10}, 0x40001e3790, 0x400076ef90, 0x400007f200, {0x10da520, 0x4000c8c120}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/schema.go:678 +0x298
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0x4000319b20?, {0x17ee890?, 0x400076eb10?}, 0x40001e3790, 0x400076ef90?, {0x10da520?, 0x4000c8c120?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:917 +0x9c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0x400037c348, {0x17ee890?, 0x400076ea50?}, 0x400091ca50)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:744 +0x80c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.PlanResourceChange({0x400021f4a0, 0x400021f500, {0x4000a5e760, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.8.0/tf5muxserver/mux_server_PlanResourceChange.go:27 +0xdc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0x40002161e0, {0x17ee890?, 0x4000705c20?}, 0x400015e9a0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:786 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0x1452e00, 0x40002161e0}, {0x17ee890, 0x4000705c20}, 0x400015e930, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:404 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x4000170000, {0x17f59a0, 0x400024a1a0}, 0x40005f4900, 0x4000b61170, 0x2695888, 0x0)
	google.golang.org/grpc@v1.56.3/server.go:1335 +0xb90
google.golang.org/grpc.(*Server).handleStream(0x4000170000, {0x17f59a0, 0x400024a1a0}, 0x40005f4900, 0x0)
	google.golang.org/grpc@v1.56.3/server.go:1712 +0x7ac
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.56.3/server.go:947 +0xac
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 58
	google.golang.org/grpc@v1.56.3/server.go:958 +0x14c

Error: The terraform-provider-datadog_v3.49.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

A service definition set up to provide the metadata for our service. We had set it up for the 'staging' environment in a module because have multiple environments across multiple services and this lower environment is common among all of our services. Tried removing the count and got the same error as well.

### Actual Behavior

Breakage

### Steps to Reproduce

terraform init
terraform apply

### Important Factoids

running on aws

### References

_No response_

--- a/datadog/resource_datadog_service_definition_yaml.go
+++ b/datadog/resource_datadog_service_definition_yaml.go
@@ -178,7 +178,7 @@ func prepServiceDefinitionResource(attrMap map[string]interface{}) map[string]in
 			sort.SliceStable(sortedContacts, func(i, j int) bool {
 				attrs := []string{"type", "contact", "name"}
 				for _, attr := range attrs {
-					v := (sortedContacts[i], sortedContacts[j], attr)
+					v := compareNullableStringArg(sortedContacts[i], sortedContacts[j], attr)
 					if v != nil {
 						return *v
 					}

--- a/datadog/resource_datadog_service_definition_yaml.go
+++ b/datadog/resource_datadog_service_definition_yaml.go
@@ -121,6 +121,20 @@ func flattenYAMLToString(input map[string]interface{}) (string, error) {
 	return string(result), nil
 }
 
+func compareNullableStringArg(left map[string]interface{}, right map[string]interface{}, key string) *bool {
+	valueLVal, valueLOk := left[key]
+	valueRVal, valueROk := right[key]
+	if valueLVal != valueRVal && valueLOk && valueROk {
+		lv, lok := valueLVal.(string)
+		rv, rok := valueRVal.(string)
+		if lok && rok {
+			v := lv == rv
+			return &v
+		}
+	}
+	return nil
+}
+
 func prepServiceDefinitionResource(attrMap map[string]interface{}) map[string]interface{} {
 	if isBackstageSchema(attrMap) {
 		// Don't prepare 3rd party schemas
@@ -161,22 +175,13 @@ func prepServiceDefinitionResource(attrMap map[string]interface{}) map[string]in
 			for _, contact := range contacts {
 				sortedContacts = append(sortedContacts, contact.(map[string]interface{}))
 			}
-
 			sort.SliceStable(sortedContacts, func(i, j int) bool {
-				typeLVal, typeLOk := sortedContacts[i]["type"]
-				typeRVal, typeROk := sortedContacts[j]["type"]
-				if typeLVal != typeRVal && typeLOk && typeROk {
-					return typeLVal.(string) < typeRVal.(string)
-				}
-				contactLVal, contactLOk := sortedContacts[i]["contact"]
-				contactRVal, contactROk := sortedContacts[j]["contact"]
-				if contactLVal != contactRVal && contactLOk && contactROk {
-					return contactLVal.(string) < contactRVal.(string)
-				}
-				nameLVal, nameLOk := sortedContacts[i]["name"]
-				nameRVal, nameROk := sortedContacts[j]["name"]
-				if nameLOk && nameROk {
-					return nameLVal.(string) < nameRVal.(string)
+				attrs := []string{"type", "contact", "name"}
+				for _, attr := range attrs {
+					v := (sortedContacts[i], sortedContacts[j], attr)
+					if v != nil {
+						return *v
+					}
 				}
 				return false
 			})
@@ -232,7 +237,9 @@ func normalizeArrayField(attrMap map[string]interface{}, key string) {
 func getNameField(data interface{}) string {
 	if stringMap, ok := data.(map[string]interface{}); ok {
 		if name, ok := stringMap["name"]; ok {
-			return name.(string)
+			if v, ok := name.(string); ok {
+				return v
+			}
 		}
 	}
 	return ""

